### PR TITLE
fix(stepfunctions-tasks): missing tags & perms for emr cluster creation

### PR DESCRIPTION
**Pull request by msambol**
Per the [documentation](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-managed-iam-policies.html#manually-tagged-resources):

* To use managed policies, pass the user tag `for-use-with-amazon-emr-managed-policies = true` when you provision a cluster with the CLI, SDK, or another method.
* For resources that are not created by Amazon EMR, you must add tags to those resources. For example, you must tag Amazon EC2 subnets, EC2 security groups (if not created by Amazon EMR), and VPCs (if you want Amazon EMR to create security groups).

Also, `AmazonEMRServicePolicy_v2` only has `iam:PassRole` on the default EMR roles and needs this on the cluster role created by the CDK.

Running the step function:

Screenshot 2023-12-11 at 5 24 09 AM

Screenshot 2023-12-11 at 5 24 03 AM

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/emr-create-cluster.ts [hunk 1]
```diff
@@ -192,6 +193,14 @@ export class EmrCreateCluster extends sfn.TaskStateBase {
     this._serviceRole = this.props.serviceRole ?? this.createServiceRole();
     this._clusterRole = this.props.clusterRole ?? this.createClusterRole();
 
+    // Service role must be able to iam:PassRole on the cluster role
+    this._clusterRole.grantPassRole(this._serviceRole);
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/emr-create-cluster.ts:197]: `AmazonEMRServicePolicy_v2` only grants `iam:PassRole` on the default role. Need this here for the cluster role we create.

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/emr-create-cluster.ts [hunk 2]
```diff
@@ -298,7 +307,12 @@ export class EmrCreateCluster extends sfn.TaskStateBase {
 
     const policyStatements = [
       new iam.PolicyStatement({
-        actions: ['elasticmapreduce:RunJobFlow', 'elasticmapreduce:DescribeCluster', 'elasticmapreduce:TerminateJobFlows'],
+        actions: [
+          'elasticmapreduce:RunJobFlow',
+          'elasticmapreduce:DescribeCluster',
+          'elasticmapreduce:TerminateJobFlows',
+          'elasticmapreduce:AddTags',
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/emr-create-cluster.ts:314]: I added `AddTags`.

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/emr-create-cluster.ts [hunk 3]
```diff
@@ -359,11 +373,7 @@ export class EmrCreateCluster extends sfn.TaskStateBase {
   private createServiceRole(): iam.IRole {
     if (cdk.FeatureFlags.of(this).isEnabled(ENABLE_EMR_SERVICE_POLICY_V2)) {
       return new iam.Role(this, 'ServiceRole', {
-        assumedBy: new iam.ServicePrincipal('elasticmapreduce.amazonaws.com', {
-          conditions: {
-            StringEquals: { 'aws:RequestTag/for-use-with-amazon-emr-managed-policies': 'true' },
-          },
-        }),
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/emr-create-cluster.ts:366]: This condition isn't needed and was actually causing it to fail.

File: packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions-tasks/test/emr/integ.emr-create-cluster-with-v2-policy.ts [hunk 4]
```diff
@@ -1,26 +1,62 @@
...
+  },
   name: 'Cluster',
+  releaseLabel: 'emr-6.15.0',
   integrationPattern: sfn.IntegrationPattern.RUN_JOB,
 });
 
+new sfn.StateMachine(stack, 'SM', {
+  definition: step,
+});
+
```
Comment (msambol) [inline packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions-tasks/test/emr/integ.emr-create-cluster-with-v2-policy.ts:57]: this wasn't actually deploying a state machine. I added this and ran the step function.

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/emr-create-cluster.ts (hunk 3 above)
Comment (paulhcsun) [reply packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/emr-create-cluster.ts:366]: Could you explain your reasoning for why it's not needed? Also how was it causing the integ tests to fail?

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/test/emr/emr-create-cluster.test.ts [hunk 5]
```diff
@@ -1314,3 +1346,11 @@ test('Task throws if WAIT_FOR_TASK_TOKEN is supplied as service integration patt
     });
   }).toThrow(/Unsupported service integration pattern. Supported Patterns: REQUEST_RESPONSE,RUN_JOB. Received: WAIT_FOR_TASK_TOKEN/);
 });
+
+function hasTags(expectedTags: Array<{Key: string, Value: string}>) {
```
Comment (paulhcsun) [inline packages/aws-cdk-lib/aws-stepfunctions-tasks/test/emr/emr-create-cluster.test.ts:1350]: Is this being used anywhere?

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/emr-create-cluster.ts (hunk 3 above)
Comment (msambol) [reply packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/emr-create-cluster.ts:366]: When included, I get the following error:

```
EMR service role arn:aws:iam::<REDACTED>:role/aws-cdk-emr-create-cluste-EmrCreateClusterServiceRo-GrrjLUAR1a2Q is invalid
```

**Review by paulhcsun — DISMISSED** [review]
Overall looks good to me. This PR also adds additional integ tests that deploys EMR cluster from a state machine and is working. Thanks for the fix!

**Review by paulhcsun — APPROVED** [review]

**Pull request merged**